encrypter.cpp

```diff
@@ -0,0 +1,202 @@
+#include <iostream>
+#include <string.h>
+#include <fstream>
+#include <ios>
+#include <random>
+#include <sstream>
+
+
+//=====put_error=====//
+void put_error(const std::string& error_message)
+{
+  std::cout << error_message;
+  exit(1);
+}
+
+//=====shuffle=====//
+
+void shuffle(std::vector<char>& input)
+{
+  std::default_random_engine generator;
+  std::uniform_int_distribution<int> distribution(0, input.size() - 1);
+  int i = 0;
+
+  for (std::vector<char>::iterator iter = input.begin(); iter < input.end(); ++i, ++iter)
+  {
+    int random = distribution(generator);
+    char tmp = input[i];
+    input[i] = input[random];
+    input[random] = tmp;
+  } 
+}
+
+//=====replace_extension=====//
+std::string replace_extension(const std::string& filename, const std::string& new_extension)
+{
+  int i;
+  std::string new_name;
+  std::string::const_iterator iter = new_extension.begin();
+
+  for (i = 0; filename[i] != '.'; ++i);
+
+  ++i;
+  if (filename[i] == '.')
+    put_error("Oh, snap! Something went wrong.\n");
+
+  new_name = filename.substr(0, i + 1);
+
+  for (int k = 0; iter < new_extension.end(); ++k, ++i, ++iter)
+    new_name[i] = new_extension[k];
+
+  return new_name;
+}
+
+//=====read_file=====//
+std::vector<char> read_file(const std::string& filename, const std::string error_message, std::streampos& size)
+{
+  std::ifstream file (filename, std::ios::in|std::ios::binary|std::ios::ate);
+  size = file.tellg();
+  if (!file.is_open())
+  {
+    file.close();
+    put_error(error_message);
+  }
+
+  char inp[(long int)(size)];
+  file.seekg(0, std::ios::beg);
+  file.read(inp, size);
+  file.close();
+  std::vector<char> input(inp, inp + size);
+
+  if (size == 0)
+    put_error("Given file is empty\n");
+
+  return input;
+}
+
+//=====write_file=====//
+void write_file(const std::string& filename, std::vector<char> output, const std::string error_message, std::streampos& size)
+{
+  std::ofstream file (filename, std::ios::out|std::ios::binary|std::ios::trunc);
+  if (!file.is_open())
+  {
+    put_error(error_message);
+  }
+  
+  file.write(&output[0], size);
+}
+
+//=====make_key=====//
+void make_key(const std::string& filename)
+{
+  std::streampos size;
+  std::vector<char> input = read_file(filename, "Couldn't access file-base for a key.\n", size);
+  shuffle(input);
+  std::string keyname = replace_extension(filename, "key");
+  write_file(keyname, input, "Couldn't open file to save the result.\n", size);
+}
+
+//=====read_key_position=====//
+std::streampos read_key_position(const std::string keyname)
+{
+  std::streampos start = 0;
+  std::ifstream meta (keyname + ".meta");
```

Comment:
"meta_file" or even "f" is better than "meta". Because "meta" doesn't tell it's a file.